### ISSUE TITLE
Cannot set an enum as argument in services.php

### ISSUE THREAD

**Issue by ludekbenedik** [symfony/symfony#50275]
### Symfony version(s) affected

5.4.23

### Description

Although Symfony 5.4 allows set an enum as an argument in DI Definition, it is not possible set the enum as the argument in Configurator.

Works:
```php
protected function build(ContainerBuilder $container)
{
    $container->register(MyService::class)
        ->setPublic(true)
        ->setArgument('$myEnum', MyEnum::First)
    ;
}
```

Not works:
```php
return static function (ContainerConfigurator $configurator): void {
    $configurator->services()->set(MyService::class)
        ->public()
        ->arg('$myEnum', MyEnum::First)
    ;
};
```

Throws: *Cannot use values of type "App\Service\MyEnum" in service configuration files.*

### How to reproduce

config/services.php

```php
$configurator->services()->set(MyService::class)
    ->arg('$myEnum', MyEnum::First)
;
```

src/Service/MyEnum.php

```php
<?php

namespace App\Service;

enum MyEnum: string
{
    case First = 'first';
    case Second = 'second';
}
```

src/Service/MyService.php

```php
<?php

namespace App\Service;

class MyService
{
    public function __construct(MyEnum $myEnum)
    {
    }
}
```

### Possible Solution

Symfony\Component\DependencyInjection\Loader\Configurator\AbstractConfigurator::processValue

Line 102

Replace

```php
case null === $value:
case \is_scalar($value):
    return $value;
```

by

```php
case null === $value:
case \is_scalar($value):
case $value instanceof \UnitEnum:
    return $value;
```

**Comment by stof** [comment]
This change makes sense. Can you submit a PR with the fix (and a test preventing regressions) ?

**Comment by nicolas-grekas** [comment]
This works on 6.2 since #48045

**Comment by ludbene** [comment]
I wrote some [relevant arguments](https://github.com/symfony/symfony/pull/50330#issuecomment-1549907811) why this fix/feature should be merged to 5.4 LTS.

**Comment by stof** [comment]
*features* are **never** merged into patch versions.
